### PR TITLE
BitArray

### PR DESCRIPTION
**Pull request by will**
Adds documentation to BitArray.

Adds two methods taking advantage of the representation to toggle individual bits, and to invert the entire  array.

### CODE REVIEW

File: src/bit_array.cr [hunk 1]
```diff
@@ -1,22 +1,51 @@
+# BitArray is an array data structure that compactly stores bits.
+#
+# Bits externally represented as `Bool`s and are stored internally as
```
Comment (PragTob) [inline src/bit_array.cr]: Bits are

File: src/bit_array.cr [hunk 2]
```diff
@@ -1,22 +1,51 @@
...
+# `UInt32`s. The total number of bits stored is set at creation and is
+# immutable.
+#
+# `BitArray` includes all the methods in `Enumerable`.
```
Comment (PragTob) [inline src/bit_array.cr]: dunno what kind of wording we want to use, in general I personally fancy "x behaves like a y"

**Comment by PragTob** [comment]
Looks good to me, just minor comments.

**Comment by asterite** [comment]
@will Thanks so much for this! It's really nice that you add docs as you use and enhance a class.

@PragTob Thanks for the minor comments, I'll apply them.

**Pull request merged**